src/auto/dbus_connection.rs

```diff
@@ -0,0 +1,1264 @@
+        Box_::pin(crate::GioFuture::new(self, move |obj, send| {
+            let cancellable = Cancellable::new();
+            obj.flush(Some(&cancellable), move |res| {
+                send.resolve(res);
+            });
+
+            cancellable
+        }))
+    }
+
+    pub fn flush_sync<P: IsA<Cancellable>>(
+        &self,
+        cancellable: Option<&P>,
+    ) -> Result<(), glib::Error> {
+        unsafe {
+            let mut error = ptr::null_mut();
+            let _ = gio_sys::g_dbus_connection_flush_sync(
+                self.to_glib_none().0,
+                cancellable.map(|p| p.as_ref()).to_glib_none().0,
+                &mut error,
+            );
+            if error.is_null() {
+                Ok(())
+            } else {
+                Err(from_glib_full(error))
+            }
+        }
+    }
+
+    pub fn get_capabilities(&self) -> DBusCapabilityFlags {
+        unsafe {
+            from_glib(gio_sys::g_dbus_connection_get_capabilities(
+                self.to_glib_none().0,
+            ))
+        }
+    }
+
+    pub fn get_exit_on_close(&self) -> bool {
+        unsafe {
+            from_glib(gio_sys::g_dbus_connection_get_exit_on_close(
+                self.to_glib_none().0,
+            ))
+        }
+    }
+
+    #[cfg(any(feature = "v2_60", feature = "dox"))]
+    pub fn get_flags(&self) -> DBusConnectionFlags {
+        unsafe { from_glib(gio_sys::g_dbus_connection_get_flags(self.to_glib_none().0)) }
+    }
+
+    pub fn get_guid(&self) -> Option<GString> {
+        unsafe { from_glib_none(gio_sys::g_dbus_connection_get_guid(self.to_glib_none().0)) }
+    }
+
+    pub fn get_last_serial(&self) -> u32 {
+        unsafe { gio_sys::g_dbus_connection_get_last_serial(self.to_glib_none().0) }
+    }
+
+    pub fn get_peer_credentials(&self) -> Option<Credentials> {
+        unsafe {
+            from_glib_none(gio_sys::g_dbus_connection_get_peer_credentials(
+                self.to_glib_none().0,
+            ))
+        }
+    }
+
+    pub fn get_stream(&self) -> Option<IOStream> {
+        unsafe { from_glib_none(gio_sys::g_dbus_connection_get_stream(self.to_glib_none().0)) }
+    }
+
+    pub fn get_unique_name(&self) -> Option<GString> {
+        unsafe {
+            from_glib_none(gio_sys::g_dbus_connection_get_unique_name(
+                self.to_glib_none().0,
+            ))
+        }
+    }
+
+    pub fn is_closed(&self) -> bool {
+        unsafe { from_glib(gio_sys::g_dbus_connection_is_closed(self.to_glib_none().0)) }
+    }
+
+    //pub fn register_object(&self, object_path: &str, interface_info: &DBusInterfaceInfo, vtable: /*Ignored*/Option<&DBusInterfaceVTable>, user_data: /*Unimplemented*/Option<Fundamental: Pointer>) -> Result<(), glib::Error> {
+    //    unsafe { TODO: call gio_sys:g_dbus_connection_register_object() }
+    //}
+
+    #[cfg(any(feature = "v2_46", feature = "dox"))]
+    pub fn register_object_with_closures(
```

Comment:
Same as for `bus_own_name_on_connection_with_closures`

Reply:
interestingly, the generator decided to ignore the returned ID altogether..

Reply:
Hrm... @EPashkin any guesses why? :)

Reply:
Result of very old https://github.com/gtk-rs/gir/pull/187
Can overridden with
```toml
[[object.function]]
name = "register_object_with_closures"
[object.function.return]
use_return_for_result = true
```